ext/liquid_c/tokenizer.c

```diff
@@ -105,14 +117,19 @@ void tokenizer_next(tokenizer_t *tokenizer, token_t *token)
                     goto found;
                 }
                 token->type = TOKEN_VARIABLE;
+                if(token->str[cursor - tokenizer->cursor - 3] == '-')
+                    token->rstrip = tokenizer->lstrip_flag = 1;
                 goto found;
             }
             // unterminated variable
-            cursor = tokenizer->cursor + 2;
+            cursor = tokenizer->cursor + 3;
+            tokenizer->lstrip_flag = 0;
             goto found;
         }
     }
     cursor = last + 1;
+    if(tokenizer->lstrip_flag)
+        token->lstrip = 1;
```

Comment:
This can also unconditionally do `token->lstrip = tokenizer->lstrip_flag;`

You should follow that with `tokenizer->lstrip_flag = 0;` as well, like you did above